MFLT-2946 Add android SDK settings to bort reference, indicate caliper logs in 3.3 (#11)
memfault/memfault-docs: docs/android/android-bort.md

Metrics are collected using the
[Batterystats subsystem](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/os/BatteryStats.java).

> Logs are tailed from [logcat](https://developer.android.com/studio/command-line/logcat).

Reviewer: ```suggestion
Logs are collected from [logcat](https://developer.android.com/studio/command-line/logcat) around issue occurances.
```

Tailed suggests to me we've got a live logcat process running at all times.
I'd also mention we only capture around observed issues.

## Features and Configuration

The Bort SDK contains an _Over The Air_ settings system so that settings are
defined on the web in your Memfault project, and the SDK will periodically pull

> the latest settings. This makes it easy to enable or disable different data

Reviewer: ```suggestion
the latest settings at run-time. This makes it easy to enable or disable different data
```

One might think it happens (only) at build-time?

sources, or change configuration options. Settings can be updated through the web UI

> or programmatically via web endpoint -- no need to deploy an app or OS update.

Reviewer: ```suggestion
or programmatically via a web endpoint -- no need to deploy an app or OS update.
```

The settings can be found in the `Settings` part of the web UI. SDK settings are
only visible to and configurable by project administrators.

> 

Reviewer: Maybe add a screenshot?

> ### Default SDK Settings

Reviewer: Maybe call it "Fallback SDK Settings" or "Baked-in SDK Settings" or "Build-time SDK Settings"?
(In the section title and also below in the paragraph.)

I already use the term "Default Value" in the UI for the Memfault-picked, global defaults.

As part of the build process, the Bort SDK will fetch the current SDK settings and store
them as a file called `settings.json`. We recommend storing this file in your version
control system. The Bort SDK embeds this file as the default SDK settings, used temporarily

> until fresh settings are retrieved from the web.

Reviewer: ```suggestion
until fresh settings are retrieved from the web, at run-time.
```

(One might think until the next build)

If, during the build process, the Bort SDK detects the local `settings.json` is stale,
the build will intentionally fail to alert you that the default settings are out of
date. To fix this, remove the stale `settings.json` file and re-run the build.

If you wish to disable this check, you can skip this step when building the Bort APK by setting a gradle property:

```
SKIP_DOWNLOAD_SYSTEM_SETTINGS_JSON=1
```

> 

Reviewer: Below in line ~180:

> There are configuration options to set the period of this task as well as the
initial delay for when the first bug report is generated (e.g. if you wish to
wait until more data is available) is described below.

I don't see that description any where. Maybe just refer to the `[SDK settings](#features-and-configuration)` like you did elsewhere?

Author: Great catch, thanks!
